### PR TITLE
Non divisible by 16 texture resize and warning

### PR DESCRIPTION
**Pull request by Jupeyy**

### CODE REVIEW

File: src/engine/client/graphics_threaded.cpp [hunk 1]
```diff
@@ -448,6 +448,23 @@ IGraphics::CTextureHandle CGraphics_Threaded::LoadTextureRaw(int Width, int Heig
 	if((Flags&IGraphics::TEXLOAD_NO_2D_TEXTURE) != 0)
 		Cmd.m_Flags |= CCommandBuffer::TEXFLAG_NO_2D_TEXTURE;
 
+	if((Flags&IGraphics::TEXLOAD_TO_2D_ARRAY_TEXTURE) != 0 || (Flags&IGraphics::TEXLOAD_TO_3D_TEXTURE) != 0)
+	{
+		if(Width == 0 || (Width % 16) != 0 || Height == 0 || (Height % 16) != 0)
+		{
+			SGraphicsWarning NewWarning;
+			NewWarning.m_WarningMsg = std::string("The width and height of texture");
```
Comment (edg-l) [inline src/engine/client/graphics_threaded.cpp:456]: do we use std string now? Don't we always use a buffer + str_cat?

File: src/engine/client/backend_sdl.cpp [hunk 2]
```diff
@@ -2253,11 +2273,31 @@ void CCommandProcessorFragment_OpenGL3_3::Cmd_Texture_Create(const CCommandBuffe
 				p3DImageData = (uint8_t*)malloc(ImageColorChannels * Width * Height);
 			int Image3DWidth, Image3DHeight;
 
-			if(IsSingleLayer || (Width != 0 && Width % 16 == 0 && Height != 0 && Height % 16 == 0 && Texture2DTo3D(pTexData, Width, Height, ImageColorChannels, 16, 16, p3DImageData, Image3DWidth, Image3DHeight)))
+			int ConvertWidth = Width;
```
Comment (edg-l) [inline src/engine/client/backend_sdl.cpp:2276]: This code looks duplicated

Comment (Jupeyy) [reply src/engine/client/backend_sdl.cpp:2276]: yeah, fixed when cleaning up the whole opengl class differences

**Comment by heinrich5991** [comment]
bors r+

Thanks for the fixes.

**Comment by BannZay** [comment]
@heinrich5991, @Jupeyy 

I think that the warning is in a wrong place. It might be helpful if this warn would pop out on save/load map in editor or in server console on map change. But end users can do nothing with such warning

**Comment by BannZay** [comment]
I guess its not a good practice to force people to report about problems this way. Sorry but I miss alternative ways of it being usefull

**Comment by Jupeyy** [comment]
its required so people with broken entities notice that their entities are broken, a silent warning would be too less

**Comment by BannZay** [comment]
If map resource is broken - user can do nothing about it

**Comment by Jupeyy** [comment]
they can contact the server admin or disable warnings

**Comment by BannZay** [comment]
why would not show this error to admin only then ? And I dont like to disable all warnings because of unwanted behavior of this one. Well, maybe I underestimate our users but I see only problems behind this warning for endusers

**Comment by Jupeyy** [comment]
when the user sees this warning, the server admin already ignored the warning

**Comment by def-** [comment]
We also have this warning in map editor btw.

**Comment by BannZay** [comment]
Can not get this warning (I use dark moon. I press "Load current map", edit map and then "Save")
I see no warnings

**Comment by Jupeyy** [comment]
yeah it will be fixed with #2691
for that map

**Comment by def-** [comment]
and I just went through all maps on DDNet and fixed them.

**Comment by BannZay** [comment]
I mean I see warning ingame but not in editor. Hope I will be able to see this warning in editor.

But still, I would not bother endusers with such problems so hard

**Comment by Jupeyy** [comment]
i think have more from a warning telling them that there might be visual bugs, then just see broken maps/entities

**Comment by Learath2** [comment]
The issue that triggered this change wasn't mapres being broken. It was users using broken custom entities. The warning indeed would tell them what is broken and they can fix it.

**Comment by BannZay** [comment]
@Learath2 yes, I would like to receive this warning on client start if MY entities is broken
If map is broken - I dont care

The irony of this is that the broken entities most likely won't bother them as much as this huge warning with OK button

**Comment by Jupeyy** [comment]
entities only load when being used tho
about bothering


it will look like this for everyone, who'd use this broken entities(as an example), even when not zoomed in, very noticable

!(this)[https://user-images.githubusercontent.com/6654924/91368169-b0188200-e808-11ea-8a06-7eb887e16275.png]

**Pull request merged**